src/Security/XpubTokenGenerator.php

```diff
@@ -18,13 +18,14 @@ public function __construct(string $clientId, string $clientSecret)
         $this->clientSecret = $clientSecret;
     }
 
-    public function generate(UserInterface $user) : string
+    public function generate(UserInterface $user, bool $newSession = false) : string
     {
         return JWT::encode([
             'iss' => $this->clientId,
             'iat' => time(),
             'exp' => time() + self::TOKEN_TTL,
             'id' => $user->getUsername(),
+            'new-session' => $newSession,
```

Comment:
boolean parameters in protocols are not stable as they tend to be extended, would prefer `'session' => 'new'|'existing'` etc. allowing new values if need be, with backward compatibility

Reply:
xPub change has already been merged IIRC, we can change in the future if we do need to extend, as the property can be renamed to lose the `new-` prefix.